src/parser/shared/modules/CastEfficiency.js

```diff
@@ -106,6 +106,38 @@ class CastEfficiency extends Analyzer {
     return timeSpentCasting;
   }
 
+  _getTimeSpentOnGcd(spellId) {
+    const ability = this.abilities.getAbility(spellId);
+
+    if (ability) {
+      const cdInfo = this._getCooldownInfo(ability);
+
+      let casts;
+      if (ability.castEfficiency.casts) {
+        casts = ability.castEfficiency.casts(this.abilityTracker.getAbility(spellId), this.owner);
+      } else {
+        casts = cdInfo.casts;
+      }
+
+      const averagetimeSpentOnAbility = this._getTimeSpentCasting(ability) / casts;
+      // If the abilities GCD is longer than the time spent casting, use the GCD
+      if (ability.gcd && averagetimeSpentOnAbility < ability.gcd.base) {
+        return ability.gcd.base * casts;
```

Comment:
To be honest, I don't think I know enough to make this 100% accurate. I've added a gcd modifier based on the ending haste of the encounter. This will make it more accurate than it was before, but it will still have a margin of error. I don't know what I would need to do short of creating a parser to log each cast.

Reply:
This can also be inconsistent if the fight ended with something like BL. It’s not uncommon for people to blow CDs during the last few percents.

I think we’d indeed need an analyzer that tracks time spent on gcds, but even that is non trivial due to gcds scaling with haste changes.